snapcraft/plugins/python.py

```diff
@@ -438,17 +439,13 @@ def list(self, exec_func=None):
         """Return a dict of installed python packages with versions."""
         if not exec_func:
             exec_func = self._exec_func
-        cmd = [*self._runnable, 'list']
+        cmd = [*self._runnable, 'list', '--format=json']
 
         output = exec_func(cmd, env=self._env)
-        package_listing = {}
-        version_regex = re.compile('\((.+)\)')
-        for line in output.splitlines():
-            line = line.split()
-            m = version_regex.search(line[1])
-            package_listing[line[0]] = m.group(1)
-
-        return package_listing
+        return {
+            package['name']: package['version']
+            for package in json.loads(output)
+        }
```

Comment:
We are testing that --json returns a list of packages just in the format we require, which is the only thing I'm after here, because it enables me to rely on that for the next branch about asset tracking.

We are not testing anything related to the local config file. And I'm ok with that because the config affects many more things than the output of pip list. As I mentioned on the bug, that's a separate and bigger problem that should probably be solved with your suggestion about the environment variable.

The solution for the slow integration tests is to run them in parallel. Running lxc in travis will allow us to start working on that.